File: code/10/solution.go
package main

import "fmt"

const upper = 2000000

var p = [upper]bool{}

func main() {
	sum := int64(0)
	for i := int64(2); i < upper; i++ {
		if p[i] {
			continue
		}
		sum += i
		for j := i << 1; j < upper; j += i {
			p[j] = true
		}
	}
	fmt.Printf("%d\n", sum)
}

File: code/10/solution.py
from random import random
from math import *

upper = 2000000
p = [False] * upper

sum = 0
for i in range(2, upper):
    if p[i]:
        continue
    sum += i
    for j in range(i << 1, upper, i):
        p[j] = True

print(sum)
